src/sentry/monitors/serializers.py

```diff
@@ -11,7 +11,59 @@
 from sentry.monitors.validators import IntervalNames
 
 from ..models import Environment
-from .models import Monitor, MonitorCheckIn, MonitorEnvironment, MonitorStatus
+from .models import (
+    Monitor,
+    MonitorCheckIn,
+    MonitorEnvBrokenDetection,
+    MonitorEnvironment,
+    MonitorIncident,
+    MonitorStatus,
+)
+
+
+class MonitorEnvBrokenDetectionSerializerResponse(TypedDict):
+    userNotifiedTimestamp: datetime
+    environmentMutedTimestamp: datetime
+
+
+@register(MonitorEnvBrokenDetection)
+class MonitorEnvBrokenDetectionSerializer(Serializer):
+    def serialize(self, obj, attrs, user, **kwargs) -> MonitorEnvBrokenDetectionSerializerResponse:
+        return {
+            "userNotifiedTimestamp": obj.user_notified_timestamp,
+            "environmentMutedTimestamp": obj.env_muted_timestamp,
+        }
+
+
+class MonitorIncidentSerializerResponse(TypedDict):
+    startingTimestamp: datetime
+    resolvingTimestamp: datetime
+    brokenNotice: MonitorEnvBrokenDetectionSerializerResponse | None
+
+
+@register(MonitorIncident)
+class MonitorIncidentSerializer(Serializer):
+    def get_attrs(
+        self, item_list: Sequence[Any], user: Any, **kwargs: Any
+    ) -> MutableMapping[Any, Any]:
+        broken_detections = list(
+            MonitorEnvBrokenDetection.objects.filter(monitor_incident__in=item_list)
+        )
+        serialized_broken_detections = {
+            detection.monitor_incident_id: serialized
+            for serialized, detection in zip(serialize(broken_detections, user), broken_detections)
+        }
+        return {
+            incident: {"broken_detection": serialized_broken_detections[incident.id]}
```

Comment:
We should use `.get(incident.id)` here since we're not guaranteed a broken detection row, afaik?

Reply:
ahh shoot yes, should write that test, it would've caught this too haha